app/controllers/dishes_controller.rb

```diff
@@ -0,0 +1,12 @@
+class DishesController < ApplicationController
+  def show
+    @dish = Dish.find(params[:id])
+  end
+
+  def add_ingredient
+    @dish = Dish.find(params[:id])
+    @dish.ingredients << Ingredient.find(params[:dish][:ingredient_ids])
+    redirect_to dish_path(@dish)
+  end
```

Comment:
This method doesn't follow RESTful practices. Going through your implementation it looks like you want to create an association between a dish and an ingredient so this should be a `create` action.

The location of this method should also be in the child controller or the join controller, but almost never in the parent controller.

```ruby
class IngredientsController < ApplicationController
  def create
    @dish = Dish.find(params[:id])
    @dish.ingredients << Ingredient.find(params[:dish][:ingredient_ids])

    redirect_to dish_path(@dish)
  end
end
```

OR

```ruby
class DishIngredientsController < ApplicationController
  def create
    @dish = Dish.find(params[:id])
    @dish.ingredients << Ingredient.find(params[:dish][:ingredient_ids])

    redirect_to dish_path(@dish)
  end
end
```